numba/core/typing/templates.py

```diff
@@ -1038,15 +1016,33 @@ class _OverloadAttributeTemplate(AttributeTemplate):
     def __init__(self, context):
         super(_OverloadAttributeTemplate, self).__init__(context)
         self.context = context
+        self._init_once()
 
-    @classmethod
-    def do_class_init(cls):
-        """
-        Register attribute implementation.
+    def _get_target_registry(self):
+        """Returns the registry for the current target
         """
-        from numba.core.imputils import lower_getattr
+        from numba.core.target_extension import (_get_local_target_checked,
+                                                 dispatcher_registry)
+        hwstr = self.metadata.get('target', 'generic')
+        target_hw = _get_local_target_checked(self.context, hwstr, "attribute")
+        # Get resgistry for the current hardware
+        disp = dispatcher_registry[target_hw]
+        tgtctx = disp.targetdescr.target_context
+        tgtctx.refresh()
+        if builtin_registry in tgtctx._registries:
+            reg = builtin_registry
+        else:
+            # Pick a registry in which to install intrinsics
+            registries = iter(tgtctx._registries)
+            reg = next(registries)
+        return reg
```

Comment:
`_OverloadAttributeTemplate` and `_IntrinsicTemplate` both contain this logic. Perhaps factor out, not necessary for this PR though.

Reply:
tracked in https://github.com/numba/numba/issues/7151

Reply:
thanks